src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
+    private Node<T> first;
+    private Node<T> last;
+
+    public int getListSize() {
+        return listSize;
+    }
+
+    public void setListSize(int listSize) {
+        this.listSize = listSize;
+    }
+
     @Override
     public void add(T value) {
+        if (getListSize() == 0) {
+            setListSize(getListSize() + 1);
+            Node<T> newNode = new Node<>(null, value, null);
+            first = newNode;
+            last = newNode;
+        } else {
+            setListSize(getListSize() + 1);
+            Node<T> newNode = new Node<>(last, value,null);
+            last.next = newNode;
+            last = newNode;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index >= 0 && index == getListSize()) {
+            add(value);
+        } else {
+            checkIndex(index);
+            setListSize(getListSize() + 1);
+            Node<T> testNode = nodesIterate(index);
+            Node<T> newNode = new Node<>(testNode.prev,value,testNode);
+            if (testNode.prev != null) {
+                testNode.prev.next = newNode;
+            }
+            if (index == 0) {
+                first = newNode;
+            }
+            newNode.next.prev = newNode;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listValue : list) {
+            add(listValue);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (nodesIterate(index) == null ? null : nodesIterate(index).item);
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldItem;
+        oldItem = nodesIterate(index).item;
+        nodesIterate(index).item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldItem;
+        Node<T> testNode = nodesIterate(index);
+        oldItem = testNode.item;
+        if (testNode.prev != null) {
+            testNode.prev.next = testNode.next;
+        }
+        if (testNode.next != null) {
+            testNode.next.prev = testNode.prev;
+        }
+        unlink(testNode);
+        setListSize(getListSize() - 1);
+        return oldItem;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> testNode = first;
+        for (int i = 0; i < getListSize(); i++) {
+            if (equals(object, testNode.item)) {
+                remove(i);
+                return true;
+            }
+            testNode = testNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return getListSize();
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return getListSize() == 0;
+    }
+
+    private Node<T> nodesIterate(int index) {
+        Node<T> testNode;
```

Comment:
do not use such naming, how to guess what it stores?
```suggestion
        Node<T> currentNode;
```